### ISSUE TITLE
https://kanso.app.medicmobile.org/repository is down

### ISSUE THREAD

**Issue by toomasp** [kanso/kanso#435]
kanso install fails with:

```bash
installing from directory .
Building version tree...
checking modules
request GET /repository/modules
request headers { Host: 'kanso.app.medicmobile.org',
  Accept: 'application/json' }
request data null
checking properties
request GET /repository/properties
request headers { Host: 'kanso.app.medicmobile.org',
  Accept: 'application/json' }
request data null
response: { headers: 
   { server: 'nginx/1.4.6 (Ubuntu)',
     date: 'Mon, 12 Feb 2018 08:43:01 GMT',
     'content-type': 'text/html',
     'content-length': '181',
     connection: 'close' },
  url: '',
  method: null,
  statusCode: 502 }
Error: 502: Bad Gateway
    at Function.exports.statusCodeError (/usr/lib/node_modules/kanso/lib/couchdb.js:258:16)
    at ClientRequest.<anonymous> (/usr/lib/node_modules/kanso/lib/couchdb.js:201:32)
    at ClientRequest.emit (events.js:160:13)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:539:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:112:17)
    at TLSSocket.socketOnData (_http_client.js:439:20)
    at TLSSocket.emit (events.js:160:13)
    at addChunk (_stream_readable.js:269:12)
    at readableAddChunk (_stream_readable.js:256:11)
    at TLSSocket.Readable.push (_stream_readable.js:213:10)
response: { headers: 
   { server: 'nginx/1.4.6 (Ubuntu)',
     date: 'Mon, 12 Feb 2018 08:43:01 GMT',
     'content-type': 'text/html',
     'content-length': '181',
     connection: 'close' },
  url: '',
  method: null,
  statusCode: 502 }
Failed
```

It has been failing like this for the past two weeks now.

**Comment by Radioreve** [comment]
We're experiencing the same problem here.
CI is broken :(

**Comment by mandric** [comment]
Pretty sure Medic has abandoned this project so we probably need to find
alternate hosting solutions or formally retire this project?

**Comment by caolan** [comment]
Current situation:

Medic kindly hosted the central repository for us after i retired the kan.so server. They're now updating their infrastructure and don't plan to continue hosting it as they no longer require it. I'm grateful they kept it running as long as they did.

Future:

I think we should retire the central repository as this project is not under active development.  In order for people to continue building projects based on Kanso I'd like us to provide a CouchDB data dump for a recent snapshot of the central repository. This should be accompanied with instructions for setting up your own local Kanso repository (it's easy to do).

@toomasp @Radioreve will that plan work for you?

**Comment by toomasp** [comment]
Yes, it will. Thank you!

**Comment by mandric** [comment]
@alxndrsn or @henokgetachew any chance we can get a data dump/export of the kanso.app.medicmobile.org site?  The only dependency the site has is CouchDB.  Last time I migrated iirc the easiest thing was to just copy the `.couch` files, but a JSON export or something could work too.  Not sure what the easiest thing on your end is.  Also if the http service is still available via IP address then I could get a copy of the data myself.

**Comment by Radioreve** [comment]
Hello,

It seems like the URL is finally responding after all, but that the certificate has expired the 7/03, so my ci build are still broken x) 

Is the endpoint being restored after all ?

**Comment by mandric** [comment]
I would assume that it's up temporarily just for us to get pull the data off.  Will do shortly.

**Comment by mandric** [comment]
It's not working for me...

```
$ curl --insecure https://kanso.app.medicmobile.org 
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>
```
```
$ curl --insecure https://kanso.app.medicmobile.org/repository
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>
```

**Comment by caolan** [comment]
Unfortunately, while preparing the database dump we discovered a security issue. See ticket: https://github.com/kanso/kanso/issues/436

I will still be able to provide a known-good backup of the repository for your own use soon.

**Comment by caolan** [comment]
Here's the latest repository data: https://caolan.org/static/files/repository.couch.gz

You can set up a local Kanso mirror by installing a local couchdb and doing something like the following: 

Warning: this will overwrite any local db named 'repository'!

```
$ gunzip repository.couch.gz
$ sudo mv repository.couch /var/lib/couchdb/{{ version }}/
$ sudo chown couchdb:couchdb /var/lib/couchdb/{{ version }}/repository.couch
$ sudo chmod 0644 /var/lib/couchdb/{{ version }}/repository.couch
```
Then, edit (or create) your `~/.kansorc` file so it includes:

```
exports.repositories = ['http://localhost:5984/repository'];
```

After that you should be able to `kanso install` from your local package repository mirror.

Let me know if you run into any problems.

**Comment by mandric** [comment]
Added another security issue related to the repo.  https://github.com/kanso/settings/issues/3

TL;DR make sure `settings` package is updated to >= 0.16.
